src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -145,38 +140,36 @@ def _get_bids_offers(market: Market, filters: Dict):
                 offer.serializable_dict() for offer in offers.values())
         return bids_list, offers_list
 
-    def _get_validated_bid_offer_match_list(
+    def _get_validated_recommendations(
             self, recommendations: List[Dict]) -> List[BidOfferMatch.serializable_dict]:
         """Return a validated list of BidOfferMatch instances."""
-        validated_records = []
-        for record in recommendations:
-            market = self.markets_mapping.get(record.get("market_id"), None)
+        validated_recommendations = []
+        for recommendation in recommendations:
+            market = self.markets_mapping.get(recommendation.get("market_id"))
             if market is None or market.readonly:
                 # The market is already finished or doesn't exist
                 raise MycoValidationException
 
             # Get the original bid and offer from the market
-            market_bid = market.bids.get(record.get("bid").get("id"), None)
-            market_offer = market.offers.get(record.get("offer").get("id"), None)
+            market_bids = [
+                market.bids.get(bid.get("id"), None) for bid in recommendation.get("bids")]
+            market_offers = [
+                market.offers.get(offer.get("id"), None) for offer in recommendation.get("offers")]
```

Comment:
```suggestion
            market_bids = [
                market.bids.get(bid.get("id")) for bid in recommendation.get("bids")]
            market_offers = [
                market.offers.get(offer.get("id")) for offer in recommendation.get("offers")]
```

Reply:
Also, can `recommendation["bids"]` be `None`? In that case, this would fail. Same applies for `recommendation["offers"]`

Reply:
A validation function should guarantee this won't happen https://gridsingularity.atlassian.net/browse/D3ASIM-3542

Reply:
In the description of the task you only mention that the validation function should contain the right keys. So it wouldn't necessarily check for `None` though (given the current description).

Reply:
@fievelk I have updated the ticket description to better reflect the requirement